test/validate/validate.go

```diff
@@ -224,3 +224,94 @@ func (v *Validator) ValidateDualStackControlPlane(ctx context.Context) error {
 
 	return nil
 }
+
+func (v *Validator) validateIPv4NodeProperties(ctx context.Context) error {
+	log.Print("Validating IPv6 Overlay Cluster Node properties")
+	nodes, err := k8sutils.GetNodeListByLabelSelector(ctx, v.clientset, nodeSelectorMap[v.os])
+	if err != nil {
+		return errors.Wrap(err, "failed to get node list")
+	}
+
+	for index := range nodes.Items {
+		nodeName := nodes.Items[index].ObjectMeta.Name
+		// check nodes status;
+		// nodes status should be ready after cluster is created
+		nodeConditions := nodes.Items[index].Status.Conditions
+		if nodeConditions[len(nodeConditions)-1].Type != corev1.NodeReady {
+			return errors.Wrapf(err, "node %s status is not ready", nodeName)
+		}
+
+		// get node labels
+		nodeLabels := nodes.Items[index].ObjectMeta.GetLabels()
+		for key := range nodeLabels {
+			if label, ok := v4OverlayNodeLabels[key]; ok {
+				log.Printf("label %s is correctly shown on the node %+v", key, nodeName)
+				if label != overlayClusterLabelName {
+					return errors.Wrapf(err, "node %s overlay label name is wrong; expected label:%s but actual label:%s", nodeName, overlayClusterLabelName, label)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (v *Validator) Validatev4OverlayControlPlane(ctx context.Context) error {
+	if err := v.validateIPv4NodeProperties(ctx); err != nil {
+		return errors.Wrap(err, "failed to validate IPv4 overlay node properties")
+	}
+
+	if err := v.validateHNSNetworkState(ctx); err != nil {
+		return errors.Wrap(err, "failed to validate IPv4 overlay HNS network state")
+	}
+
+	return nil
+}
+
+func (v *Validator) validateHNSNetworkState(ctx context.Context) error {
+	// check HNS network state only on windows nodes
+	nodes, err := k8sutils.GetNodeListByLabelSelector(ctx, v.clientset, nodeSelectorMap["windows"])
+	if err != nil {
+		return errors.Wrap(err, "failed to get node list")
+	}
+	// check windows HNS network state
+	for index := range nodes.Items {
+		pod, err := k8sutils.GetPodsByNode(ctx, v.clientset, privilegedNamespace, privilegedLabelSelector, nodes.Items[index].Name)
+		if err != nil {
+			return errors.Wrap(err, "failed to get privileged pod")
+		}
+
+		podName := pod.Items[0].Name
+		// exec into the pod to get the state file
+		result, err := k8sutils.ExecCmdOnPod(ctx, v.clientset, privilegedNamespace, podName, hnsNetworkCmd, v.config)
+		if err != nil {
+			return errors.Wrap(err, "failed to exec into privileged pod")
+		}
+
+		hnsNetwork, err := hnsNetworkState(result)
+		if err != nil {
+			return errors.Wrap(err, "failed to unmarshal hns network list")
+		}
+
+		if len(hnsNetwork) == 0 {
+			return errors.Wrap(err, "windows node does not have HNS network")
```

Comment:
`errors.Wrap` returns nil if the wrapped error variable is nil. This is incorrect since in this case you will be returning nil instead of your intended error. Please correct **all** instances of `errors.Wrap` in your PR where a nil error variable is being wrapped.